_posts/2020-11-04-solidity-ama-1-recap.md

```diff
@@ -45,7 +45,9 @@ I hope they will switch to a yul-based approach once the code generator is finis
 
 The Solidity compiler uses two different optimizer modules: The "old" optimizer that operates at opcode level and the "new" optimizer that operates on Yul IR code. The opcode-based optimizer applies simplification rules from the [list](https://github.com/ethereum/solidity/blob/develop/libevmasm/RuleList.h) to opcodes next to each other. It also combines equal code sets, removes unused code and some other things. The Yul-based optimizer is much more powerful, because it can work across function calls: In Yul it is not possible to perform arbitrary jumps, so it is for example possible to compute the side-effects of each function: If a function does not modify storage, a call to it can be swapped with a function that does. If a function is side-effect free and its result is multiplied by zero, you can remove the function call completely.
 
-There is single stage, the "constant optimizer" where the trade-off between deploy-time and run-time costs are taken into account. This stage tries to find a "better" representation of each number in the source, like ``0x10000000000`` can be encoded as ``PUSH6 0x10000000000`` (7 bytes and almost zero run-time costs), but it can also be encoded as `PUSH1 1 PUSH1 40 SHL` (5 bytes and a bit more expensive at run-time). Most of the time, the difference is not too relevant. The optimizer tries to simplify complicated expressions (which reduces both size and execution cost), but it also specializes or inlines functions. Especially function inlining is an operation that can cause much bigger code, but it is often done because it results in opportunities for more simplifications.
+In the opcode-based optimizer there is only one stage (the "constant optimizer") where the trade-off between deploy-time and run-time costs are taken into account. This stage tries to find a "better" representation of each number in the source, like ``0x10000000000`` can be encoded as ``PUSH6 0x10000000000`` (7 bytes and almost zero run-time costs), but it can also be encoded as `PUSH1 1 PUSH1 40 SHL` (5 bytes and a bit more expensive at run-time). Most of the time, the difference is not too relevant.
+
+The optimizer tries to simplify complicated expressions (which reduces both size and execution cost), but it also specializes or inlines functions. Especially function inlining is an operation that can cause much bigger code, but it is often done because it results in opportunities for more simplifications.
```

Comment:
Now that I look at it again, the second paragraph seems to make more sense on top, as a kind of introduction to this question.

I also added the missing `s` in `trade-offs`.

```suggestion
The optimizer tries to simplify complicated expressions (which reduces both size and execution cost), but it also specializes or inlines functions. Especially function inlining is an operation that can cause much bigger code, but it is often done because it results in opportunities for more simplifications.

In the opcode-based optimizer there is only one stage (the "constant optimizer") where the trade-offs between deploy-time and run-time costs are taken into account. This stage tries to find a "better" representation of each number in the source, like ``0x10000000000`` can be encoded as ``PUSH6 0x10000000000`` (7 bytes and almost zero run-time costs), but it can also be encoded as `PUSH1 1 PUSH1 40 SHL` (5 bytes and a bit more expensive at run-time). Most of the time, the difference is not too relevant.
```